packages/kotti-ui/source/kotti-comment/KtComment.vue

```diff
@@ -95,75 +104,74 @@ export default {
 		KtCommentInput,
 	},
 	props: makeProps(KottiComment.propsSchema),
-	data() {
-		return {
-			showInlineReply: false,
-			replyToText: null,
-			replyToUserId: Number,
-			isInlineEdit: false,
-			inlineMessageValue: '',
+	setup(props, { emit }) {
+		const isInlineEdit = ref(false)
+		const showInlineReply = ref(false)
+		const inlineMessageValue = ref<string | null>(null)
+		const userBeingRepliedTo = ref<UserData | null>(null)
+
+		const handleDelete = (commentId: number | string, isInline?: boolean) => {
+			emit('delete', {
+				id: commentId,
+				parentId: isInline ? props.id : null,
+			})
 		}
-	},
-	computed: {
-		styleObject() {
-			return {
-				avatar: true,
-				'avatar-selected': this.selected,
-			}
-		},
-		inlineMessage() {
-			return this.inlineMessageValue || this.message
-		},
-		actionOptions() {
-			const options = []
-			if (this.isEditable)
-				options.push({
-					label: 'Edit',
-					onClick: () => {
-						this.inlineMessageValue = this.inlineMessage
-						this.isInlineEdit = true
-					},
-					type: 'EDIT',
-				})
-			if (this.isDeletable)
-				options.push({
-					label: 'Delete',
-					onClick: () => this.handleDelete(this.id),
-					type: 'DELETE',
+		return {
+			actionOptions: computed<Array<KottiButton.Props>>(() =>
+				compact([
+					props.isEditable
+						? {
+								label: 'Edit',
+								onClick: () => {
+									if (inlineMessageValue.value === null)
+										inlineMessageValue.value = props.message
+									isInlineEdit.value = true
+								},
+								type: KottiButton.Type.PRIMARY,
+						  }
+						: null,
+					props.isDeletable
+						? {
+								label: 'Delete',
+								onClick: () => handleDelete(props.id),
+								type: KottiButton.Type.DANGER,
+						  }
+						: null,
+				]),
+			),
+			cancelInlineEdit: () => {
+				inlineMessageValue.value = null
+				isInlineEdit.value = false
+			},
+			handleDelete,
+			handleEditConfirm: () => {
+				isInlineEdit.value = false
+				if (!inlineMessageValue.value) return
+				emit('edit', {
+					id: props.id,
+					message: inlineMessageValue.value,
 				})
-			return options
-		},
-	},
-	methods: {
-		cancelInlineEdit() {
-			this.inlineMessageValue = ''
-			this.isInlineEdit = false
-		},
-		handleInlineReplyClick(user) {
-			this.showInlineReply = true
-			this.replyToUserId = user.userId
-			this.replyToText = `Reply to ${user.userName}`
-		},
-		handleDelete(event, type) {
-			this.$emit('delete', {
-				id: event,
-				parentId: type ? this.id : null,
-			})
-		},
-		handleInlineSubmit(event) {
-			this.showInlineReply = false
-			this.$emit('submit', event)
-		},
-		handleEditConfirm() {
-			this.isInlineEdit = false
-			if (!this.inlineMessageValue) return
-			this.$emit('edit', {
-				id: this.id,
-				message: this.inlineMessageValue,
-			})
-		},
+			},
+			handleInlineReplyClick: ({ userId, userName }: UserData) => {
+				showInlineReply.value = true
+				userBeingRepliedTo.value = { userId, userName }
```

Comment:
no need to destruct the argument, if we are passing it down as is 

arg can become `replyParentUser: UserData`
and assignment can become `userBeingRepliedTo.value = replyParentUser`

Reply:
or however you want to name the argument